experimental/incremental/queries/report.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package queries
+
+import "github.com/bufbuild/protocompile/experimental/report"
+
+// Values for [report.Report].Stage, which determine how diagnostics generated
+// across parts of the compiler are sorted.
+const (
+	stageFile int = iota
```

Comment:
Could "stage" be made into a named type (like an enum), and then these values moved into the package that defines them?

If not, perhaps a better name for the field in `report.Report` would be named more generically, like "SortOrder".

Also, if we ever allowed custom diagnostics to run as part of a compilation, we might want room between these stages, into which custom diagnostics might be sorted/interleaved. So maybe `iota*10`?

Reply:
Let's do `SortOrder`